File: CF/879B.rb
n, k = gets.split.map &:to_i
b, c = 0, 0
gets.split.map(&:to_i).each{|v|
	c = b > 0 ? (b > v ? c + 1 : 1) : 0
	b = [b, v].max
	c == k && break
}
p b

File: CF/879B.py
n, k = map(int, raw_input().split())
b, c = 0, 0
for v in map(int, raw_input().split()):
	if b > 0: c = [1, c + 1][b > v]
	b = max(b, v)
	if c == k: break
print b